### DISCUSSION TITLE
[V3] Livewire creates dynamic properties , which is deprecated in php 8.2

### DISCUSSION THREAD

**Discussion by yassinebenaid** [livewire/livewire#6464] [7. Bugs (archived)] [answered]
### Livewire version

3.0@beta.11

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. create a new component
2. use it as a full page component by assigning it to a route and visit that page
3. using any debug tool , like [laravel debugbar](https://github.com/barryvdh/laravel-debugbar) , inspect the warnings section
4. you will see a warning that looks like this : 
```bash
LOG.warning: Creation of dynamic property Livewire\Features\SupportPageComponents\LayoutConfig::$viewContext is deprecated in /var/www/html/vendor/livewire/livewire/src/Features/SupportPageComponents/SupportPageComponents.php on line 101
```

### Additional information

this is the code i got in messages section using  [laravel debugbar](https://github.com/barryvdh/laravel-debugbar)
```bash
LOG.warning: Creation of dynamic property Livewire\Features\SupportPageComponents\LayoutConfig::$viewContext is deprecated in /var/www/html/vendor/livewire/livewire/src/Features/SupportPageComponents/SupportPageComponents.php on line 101
```

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment, marked as answer]
@yassinebenaid thanks for reporting! PRs to fix are welcome! 🙂

**Comment by tristanisham** [comment]
For anyone else who finds this looking for a quick solution, add the `AllowDynamicProperties` attribute to the class Livewire is generating dynamic properties on.

```php
use \AllowDynamicProperties;

#[AllowDynamicProperties]
class Show extends Component {}
```

This will silence the warning.
